crates/solver/src/settlement_rater.rs

```diff
@@ -311,136 +311,203 @@ impl ScoreCalculator {
-        probability_success: BigRational,
-        cost_fail: BigRational,
-    ) -> BigRational {
-        tracing::trace!(
-            ?score_reference,
-            ?objective,
-            ?probability_success,
-            ?cost_fail,
-            "Computing payoff"
-        );
-        let probability_fail = BigRational::one() - probability_success.clone();
-        let payoff_success = min(objective - score_reference.clone(), self.score_cap.clone());
-        tracing::trace!(?payoff_success, "Payoff success");
-        let payoff_fail = -min(score_reference, self.score_cap.clone()) - cost_fail;
-        tracing::trace!(?payoff_fail, "Payoff fail");
-        let payoff_expectation =
-            probability_success * payoff_success + probability_fail * payoff_fail;
-        tracing::trace!(?payoff_expectation, "Payoff expectation");
-        payoff_expectation
+fn compute_optimal_bid(
+    objective: BigRational,
+    probability_success: BigRational,
+    cap: BigRational,
+) -> Result<BigRational> {
+    tracing::trace!(
+        ?objective,
+        ?probability_success,
+        ?cap,
+        "Computing optimal bid"
+    );
+    let probability_fail = BigRational::one() - probability_success.clone();
+
+    // Solvers are paid the difference between their eventually achieved objective
+    // value and the second best reported bid (aka reference score). If a solver
+    // fails to submit a solution the achieved objective is 0 and therefore they are
+    // penalized by the reference score. The optimal bidding strategy requires
+    // solvers to bid in a way that even in the worst case (i.e. they win, but
+    // the runners up bids the same value) their expected profit is zero (due to
+    // the second-price mechanism, truthful revelation of your absolute limit,
+    // i.e. 0 profit in the worst case, is the dominant strategy and will lead to
+    // optimal payouts in the average case).
+    //
+    // To do so, they discount their bid by the probability of success. This
+    // will ensure that the expected penalties (p_fail * bid) equal the
+    // expected profits (p_success * (obj - bid)).
+    //
+    // !!! Importantly, what we are trying to ensure is that
+    // => p_fail * bid = p_success * (obj - bid)
+    //
+    // which can be rewritten to:
+    let mut bid = probability_success.clone() * objective.clone();
+
+    // Now, this strategy would be optimal if rewards weren't capped.
+    // Given the cap, above equality becomes:
+    //
+    // => p_fail * std::cmp::min(bid, cap) = p_success * std::cmp::min(obj - bid,
+    // cap)
+    //
+    // We now look at the four cases that can arise from
+    // choosing the left or right argument in the `min` function on both
+    // sides of the equation:
+    if bid <= cap && objective.clone() - bid.clone() <= cap {
+        //Cap doesn't matter, equation reduces to the original one. We are done
+    } else if bid <= cap && objective.clone() - bid.clone() > cap {
```

Comment:
The argument is not so simple, see [Haris' comment](https://github.com/cowprotocol/services/pull/1730#issue-1835206186) for correct case:
```suggestion
    } else if objective.clone() - bid.clone() > cap && probability_success < 0.5 {
```
This includes all the cases of you statement but also more cases, e.g. where `bid > cap && objective.clone() - bid.clone() > cap`.
One of the current unit tests should cover this.